tests/test_printer/test_schema_directives.py

```diff
@@ -66,14 +79,35 @@ class Input:
     @strawberry.type(directives=[SensitiveData(reason="GDPR")])
     class User:
         first_name: str
+        phone: str = strawberry.field(
+            directives=[
+                SensitiveData(
+                    reason="PRIVATE",
+                    meta=[
+                        SensitiveValue(
+                            key="can_share_field", value="phone_share_accepted"
+                        )
+                    ],
+                )
+            ]
+        )
+        phone_share_accepted: bool
 
     @strawberry.type
     class Query:
         @strawberry.field
         def user(self, input: Input) -> User:
-            return User(first_name=input.first_name)
+            return User(
+                first_name=input.first_name,
+                phone="+551191551234",
+                phone_share_accepted=False,
+            )
 
     expected_type = """
+    directive @sensitiveData(reason: String!, meta: [SensitiveValue!]) on OBJECT | FIELD_DEFINITION
+
+    directive @sensitiveInput(reason: String!, meta: [SensitiveValue!]) on INPUT_OBJECT
```

Comment:
was there a particular reason for adding the schema directives to the schema? just curious 😊

Reply:
@patrick91 AFAIK (I might be wrong =P) the schema directives should be printed to the schema, shouldn't them?

But the main reason here is to allow those to be parsed by external libs, and when they find a field/type/etc "decorated" by a schema directive, they can look at the definition and understand what they mean exactly. In this example exactly, the field is defined as `phone: String! @sensitiveData(reason: "PRIVATE", meta: [{key: "can_share_field", value: "phone_share_accepted"}])`. Without knowing definition it would be impossible to know the real type for `key` and `value` (in this case they are actually `String`, but they could be an enum for example)

Reply:
yes, the reason I'm asking is that for example apollo federation doesn't need them to be printed so I thought it wouldn't be necessary for most use cases 😊

are you doing any work with schema directives and external tools?

Reply:
I'm actually already using it for a while now :)

Basically I created a custom [schema directive extension](https://github.com/blb-ventures/strawberry-django-plus/blob/main/strawberry_django_plus/directives.py#L103) to be able to run schema directives at runtime.

I use it mainly to [check permissions](https://github.com/blb-ventures/strawberry-django-plus/blob/main/strawberry_django_plus/permissions.py) (i.e. the user is logged in, the user is a superuser or the user has a given django permission globally or in the returned object). And since those directives get printed to the schema, my frontend partners parse it to know in runtime if the user can or cannot access some field, allowing them to dynamically generate queries/forms/etc. For that to work I had to monkey-patch the printerpy module in `strawberry-django-plus`, which is basically a simpler version of what I ended up suggesting here =P

Btw, that permissioning thing is a simple use case, I was inspired by the approach mentioned in places like:

* https://gqlgen.com/reference/directives/
* https://blog.logrocket.com/a-complete-guide-to-permissions-in-a-graphql-api/
* https://www.graphql-tools.com/docs/schema-directives#enforcing-access-permissions